RecoJets/JetProducers/plugins/ECFAdder.cc

```diff
@@ -0,0 +1,75 @@
+#include "RecoJets/JetProducers/interface/ECFAdder.h"
+#include "fastjet/PseudoJet.hh"
+#include "fastjet/ClusterSequence.hh"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+ECFAdder::ECFAdder(const edm::ParameterSet& iConfig) :
+  src_(iConfig.getParameter<edm::InputTag>("src")),
+  src_token_(consumes<edm::View<reco::Jet>>(src_)),
+  Njets_(iConfig.getParameter<std::vector<unsigned> >("Njets")),
+  beta_(iConfig.getParameter<double>("beta"))
+{
+  for ( std::vector<unsigned>::const_iterator n = Njets_.begin(); n != Njets_.end(); ++n )
+    {
+      std::ostringstream ecfN_str;
+      ecfN_str << "ecf" << *n;
```

Comment:
this didn't work for me:
std::string ecfN_str="ecf" + *n;
produces<edm::ValueMap<float> >(ecfN_str.c_str());
so I didn't change it.
This is however only done in the constructor.